File: PiDevWebFinal/src/Entity/Reclamation.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;
use Captcha\Bundle\CaptchaBundle\Validator\Constraints as CaptchaAssert;

/**
 * Reclamation
 *
 * @ORM\Table(name="reclamation", indexes={@ORM\Index(name="fk_idclientreclam", columns={"idclient_rec"})})
 * @ORM\Entity
 */
class Reclamation
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_reclamation", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     * @Groups("reclamation")
     */
    private $id_reclamation;
    /**
     *  @Assert\NotBlank(message="Captcha code obligatoire")
     * @CaptchaAssert\ValidCaptcha(
     *      message = "
                         La validation CAPTCHA a échoué, réessayez."
     * )
     *
     */
    protected $captchaCode;





    /**
     * @var string
     *
     * @ORM\Column(name="nom_user", type="string", length=20, nullable=true)
     *@Groups("reclamation")
     */
    private $nomUser;

    /**
     * @var string
     *
     * @ORM\Column(name="prenom_user", type="string", length=20, nullable=true)
     * @Groups("reclamation")
     */
    private $prenomUser;

    /**
     * @var string
     *
     * @ORM\Column(name="numtel", type="string", length=20, nullable=true)
     * @Groups("reclamation")
     */
    private $numtel;

    /**
     * @var string
     *
     * @ORM\Column(name="email", type="string", length=75, nullable=true)
     * @Groups("reclamation")
     */
    private $email;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="string", length=75, nullable=false)
     * @Assert\NotBlank(message="Veuillez donner les détails de votre réclamation")
     * @Groups("reclamation")
     */
    private $description;

    /**
     * @var string
     *
     * @ORM\Column(name="objet", type="string", length=20, nullable=false)
     * @Assert\NotBlank(message="Le champ objet est obligatoire")
     * @Groups("reclamation")
     */
    private $objet;

    /**
     * @var string|null
     *
     * @ORM\Column(name="etat", type="string", length=20, nullable=true, options={"default"="En attente"})
     * @Groups("reclamation")
     */
    private $etat = 'En attente';

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date", type="date", nullable=false)
     * @Groups("reclamation")
     */
    private $date;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_traitement", type="date", nullable=true)
     * @Groups("reclamation")
     */
    private $dateTraitement;

    /**
     * @var Utilisateur
     *
     * @ORM\ManyToOne(targetEntity="Utilisateur",inversedBy="reclamation")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idclient_rec", referencedColumnName="id")
     * })
     */
    private $idclientRec;

    /**
     * Reclamation constructor.
     */
    public function __construct(Utilisateur $user)
    {
        $this->date= new \DateTime('now');
        $this->nomUser=$user->getNom();
        $this->prenomUser=$user->getPrenom();
        $this->email=$user->getEmail();
        $this->numtel='+216'.$user->getNumtel();
        $this->idclientRec=$user;
    }

    /**
     * @return int
     */
    public function getIdReclamation(): int
    {
        return $this->id_reclamation;
    }

    /**
     * @param int $idReclamation
     */
    public function setIdReclamation(int $idReclamation): void
    {
        $this->id_reclamation = $idReclamation;
    }

    /**
     * @return string
     */
    public function getNomUser(): ?string
    {
        return $this->nomUser;
    }

    /**
     * @param string $nomUser
     */
    public function setNomUser(string $nomUser): void
    {
        $this->nomUser = $nomUser;
    }

    /**
     * @return string
     */
    public function getPrenomUser(): ?string
    {
        return $this->prenomUser;
    }

    /**
     * @param string $prenomUser
     */
    public function setPrenomUser(string $prenomUser): void
    {
        $this->prenomUser = $prenomUser;
    }

    /**
     * @return string
     */
    public function getNumtel(): ?string
    {
        return $this->numtel;
    }

    /**
     * @param string $numtel
     */
    public function setNumtel(string $numtel): void
    {
        $this->numtel = $numtel;
    }

    /**
     * @return string
     */
    public function getEmail(): ?string
    {
        return $this->email;
    }

    /**
     * @param string $email
     */
    public function setEmail(string $email): void
    {
        $this->email = $email;
    }

    /**
     * @return string
     */
    public function getDescription(): ?string
    {
        return $this->description;
    }

    /**
     * @param string $description
     */
    public function setDescription(string $description): void
    {
        $this->description = $description;
    }

    /**
     * @return string
     */
    public function getObjet(): ?string
    {
        return $this->objet;
    }

    /**
     * @param string $objet
     */
    public function setObjet(string $objet): void
    {
        $this->objet = $objet;
    }

    /**
     * @return string|null
     */
    public function getEtat(): ?string
    {
        return $this->etat;
    }

    /**
     * @param string|null $etat
     */
    public function setEtat(?string $etat): void
    {
        $this->etat = $etat;
    }

    /**
     * @return \DateTime
     */
    public function getDate(): ?\DateTime
    {
        return $this->date;
    }

    /**
     * @param \DateTime $date
     */
    public function setDate(?\DateTime $date): void
    {
        $this->date = $date;
    }

    /**
     * @return \DateTime|null
     */
    public function getDateTraitement(): ?\DateTime
    {
        return $this->dateTraitement;
    }

    /**
     * @param \DateTime|null $dateTraitement
     */
    public function setDateTraitement(?\DateTime $dateTraitement): void
    {
        $this->dateTraitement = $dateTraitement;
    }

    /**
     * @return Utilisateur
     */
    public function getIdclientRec(): Utilisateur
    {
        return $this->idclientRec;
    }

    /**
     * @param Utilisateur $idclientRec
     */
    public function setIdclientRec(Utilisateur $idclientRec): void
    {
        $this->idclientRec = $idclientRec;
    }

    public function getCaptchaCode()
    {
        return $this->captchaCode;
    }

    public function setCaptchaCode($captchaCode)
    {
        $this->captchaCode = $captchaCode;
    }
}

File: PiDevMobile/src/com/mycompany/entites/Reclamation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.entites;

//import java.sql.Timestamp;
//import java.time.LocalDateTime;
//import java.sql.Date;
import java.util.Comparator;
import javafx.scene.control.Button;


/**
 *
 * @author miral
 */
public class Reclamation {
    private int id_reclamation;
//    private user id;
    private String nom_user;
    private String prenom_user;
    private String numtel;
    private String email;
    private String objet;
    private String description;
    private String etat;
    
//    private Timestamp date=Timestamp.valueOf(LocalDateTime.MIN);
    private String date;
    private String date_traitement;

//    public Reclamation(Utilisateur user) {
//        this.id=user;
//    }
    
    public Reclamation() {
       
    }

    
//ajout
    public Reclamation(String objet, String description,String date) {
        this.objet = objet;
         this.description=description;
         this.date=date;
    }

//update
    public Reclamation(int id_reclamation, String nom_user, String prenom_user, String num, String email, String message, String objet, String etat,String date) {
        this.id_reclamation = id_reclamation;
//        this.id = id;
this.numtel=num;
this.nom_user = nom_user;
this.prenom_user = prenom_user;
this.email = email;
this.description = message;
this.objet=objet;
this.etat = etat;
        this.date_traitement=date;
    }
////display
//    public Reclamation(int id,String nom_user, String prenom_user, String num,String email, String message,String objet,String etat,String date) {
//        this.idReclamation=id;
//        this.nom_user = nom_user;
//        this.prenom_user = prenom_user;
//         this.numtel=num;
//        this.email = email;
//        this.description = message;
//        this.objet=objet;
//        this.etat = etat;
//        this.date=date;
//       
//    }

    public String getNumtel() {
        return numtel;
    }

    public void setNumtel(String numtel) {
        this.numtel = numtel;
    }

    public int getId_reclamation() {
        return id_reclamation;
    }

    public void setId_reclamation(int id_reclamation) {
        this.id_reclamation = id_reclamation;
    }
//
//    public user getId() {
//        return id;
//    }
//
//    public void setId(user id) {
//        this.id = id;
//    }

    public String getNom_user() {
        return nom_user;
    }

    public void setNom_user(String nom_user) {
        this.nom_user =nom_user;
    }

    public String getPrenom_user() {
        return prenom_user;
    }

    public void setPrenom_user(String prenom_user) {
        this.prenom_user =prenom_user;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email =email;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getObjet() {
        return objet;
    }

    public void setObjet(String objet) {
        this.objet = objet;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

 

    public String getDate_traitement() {
        return date_traitement;
    }

    public void setDate_traitement(String date_traitement) {
        this.date_traitement = date_traitement;
    }

 
 

    @Override
    public String toString() {
        return "Reclamation{" + "nom_user=" + nom_user + ", prenom_user=" + prenom_user + ", email=" + email + ", objet=" + objet + ", description=" + description + ", etat=" + etat + ", date=" + date + '}';
    }


    @Override
    public int hashCode() {
        int hash = 7;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Reclamation other = (Reclamation) obj;
        if (this.id_reclamation != other.id_reclamation) {
            return false;
        }
        return true;
    }
  
}
